PORT-14552 - Docs for Github ocean integration (#2494)
port-labs/port-docs: docs/build-your-software-catalog/sync-data-to-catalog/git/github-ocean/installation/installation.mdx

- Your Port user role is set to `Admin`.

## Setup

### Configure access token

To allow Port to fetch data from your GitHub Organization, you need to create an access token. Port supports two types of tokens for authentication: personal access tokens and GitHub app installation tokens.

#### Personal access token (PAT)

A Personal Access Token (PAT) is suitable if you're the only one managing the integration and don't need frequent credential rotation.  
To create a personal access token see Github's [managing your personal access tokens](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens).

- The token must belong to a user with access to the relevant Github resources (e.g., repositories, teams).

<details>
  <summary>Required permissions</summary>

  These are the necessary permissions if you're creating a fine-grained PAT.

  **Repository permissions:**
  - **Content**: Readonly
  - **Metadata**: Readonly
  - **Pull Requests**: Readonly
  
  **Organization permissions:**
  - **Webhooks**: Read and Write (for managing webhook)
</details>

#### Github app installation

> Refer to our [Github app installation guide](./github-app.mdx)

Reviewer: it is not clear if this is mandatory or not, if its a prerequisite or not, and in general i'm missing a sentence here about the app and why the user would want/need to install it

To create a personal access token see Github's [managing your personal access tokens](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens).

- The token must belong to a user with access to the relevant Github resources (e.g., repositories, teams).

<details>
  <summary>Required permissions</summary>

  These are the necessary permissions if you're creating a fine-grained PAT.

  **Repository permissions:**
  - **Content**: Readonly
  - **Metadata**: Readonly
  - **Pull Requests**: Readonly
  
  **Organization permissions:**
  - **Webhooks**: Read and Write (for managing webhook)

:::note Default permissions
You will be prompted to confirm the above listed permissions when creating a personal access token.

Permissions can be given to selected repositories in your organization, or to all repositories.   
You can reconfigure the permission at any time, giving it access to new repositories, or removing access.

:::
</details>

</TabItem>
<TabItem value="github-app" label="GitHub App">

> A GitHub App provides automatic access token rotation and is a good alternative to PATs. Refer to our [GitHub App installation guide](./github-app.mdx) to set it up.

Reviewer: ```suggestion
A GitHub App provides automatic access token rotation and is a good alternative to PATs.  
To set it up, see the [GitHub App installation guide](./github-app.mdx).
```
